app/src/main/java/com/example/sharingang/DetailedItemFragment.kt

```diff
@@ -29,14 +31,18 @@ class DetailedItemFragment : Fragment() {
     private val args: DetailedItemFragmentArgs by navArgs()
     @Inject
     lateinit var currentUserProvider: CurrentUserProvider
+    @Inject
+    lateinit var itemRepository: ItemRepository
     private val viewModel: UserProfileViewModel by viewModels()
     private val itemViewModel: ItemsViewModel by viewModels()
     private lateinit var binding: FragmentDetailedItemBinding
+    private var item: Item? = null
```

Comment:
This works, but if the fragment gets reset, because you turn the device, then this gets destroyed. Using part of the ViewModel is really what we should be doing instead.

Reply:
It was not possible to use the viewmodel here, I was getting errors in the coroutine call mentioning "addObserver" for some reason.

Reply:
But when the screen is rotated, on create is called again and it gets the item once again.

Reply:
Ah ok, that works out then